components/base/wheeled/wheeled_base.go

```diff
@@ -91,17 +100,182 @@ type wheeledBase struct {
 
-	left      []motor.Motor
-	right     []motor.Motor
-	allMotors []motor.Motor
+	left                []motor.Motor
+	right               []motor.Motor
+	allMotors           []motor.Motor
+	movementSensor      []movementsensor.MovementSensor
+	orienationSupported bool
+	orientationSensor   movementsensor.MovementSensor
+
+	opMgr  operation.SingleOperationManager
+	logger golog.Logger
+	model  referenceframe.Model
 
-	opMgr operation.SingleOperationManager
+	activeBackgroundWorkers sync.WaitGroup
 }
 
-func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
-	ctx, done := base.opMgr.New(ctx)
-	defer done()
+// createWheeledBase returns a new wheeled base defined by the given config.
+func createWheeledBase(
+	ctx context.Context,
+	deps registry.Dependencies,
+	cfg *Config,
+	baseName string,
+	logger golog.Logger,
+) (base.LocalBase, error) {
+	base := &wheeledBase{
+		widthMm:              cfg.WidthMM,
+		wheelCircumferenceMm: cfg.WheelCircumferenceMM,
+		spinSlipFactor:       cfg.SpinSlipFactor,
+		logger:               logger,
+	}
 
+	if base.spinSlipFactor == 0 {
+		base.spinSlipFactor = 1
+	}
+
+	for _, name := range cfg.Left {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "no left motor named (%s)", name)
+		}
+		positionReporting, err := m.Properties(ctx, nil)
+		if positionReporting["PositonReporting"] && err == nil {
+			base.logger.Debugf("have encoders on motor %#v", m)
+		}
+		base.logger.Debug("no encoders found on left motors")
+		base.left = append(base.left, m)
+	}
+
+	for _, name := range cfg.Right {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "no right motor named (%s)", name)
+		}
+		positionReporting, err := m.Properties(ctx, nil)
+		if positionReporting["PositonReporting"] && err == nil {
+			base.logger.Debugf("have encoders on motor %#v", m)
+		}
+		base.logger.Debug("no encoders found on right motors")
+		base.right = append(base.right, m)
+	}
+
+	for _, msName := range cfg.MovementSensor {
+		ms, err := movementsensor.FromDependencies(deps, msName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "no movement_sensor namesd (%s)", msName)
+		}
+		base.movementSensor = append(base.movementSensor, ms)
+		props, err := ms.Properties(ctx, nil)
+		if props.OrientationSupported && err == nil {
+			base.orienationSupported = props.OrientationSupported
+			base.orientationSensor = ms
+		}
+	}
+
+	base.allMotors = append(base.allMotors, base.left...)
+	base.allMotors = append(base.allMotors, base.right...)
+
+	var err error
+	base.model, err = base.createModelFrame(baseName, cfg.WidthMM)
+	if err != nil {
+		logger.Error("error creating base model, check that a name for the base and width are provided in the configuration")
+	}
+
+	return base, nil
+}
+
+func (base *wheeledBase) createModelFrame(baseName string, widthMM int) (referenceframe.Model, error) {
```

Comment:
so, without implementing a `ModelFrame` function, this doesn't implement the `ModelFramer` interface and thus isn't consumed by the frame system (unless I've misunderstood the way that works), making this dead code. But even if it was implemented, I don't think it belongs in this PR because it doesn't seem to be involved in any of the spin logic?

Reply:
That's okay. That will also be added in the future, there's already a related ticket. Will keep this function which will just be refactored into `ModelFrame`

Reply:
Will the frame system be involved in helping regulate spin in a future PR?

Reply:
maybe, or updating frame or pose states of the base.  Kind of open now but I want to set rules as to what the reference frame of the base is in an explicit way. This is all kept private so it is changeable.